python/ray/tests/test_recursion.py

```diff
@@ -6,19 +6,30 @@
 from __future__ import division
 from __future__ import print_function
 
+import pytest
+
 import ray
 
-ray.init(num_cpus=1)
 
+@pytest.fixture
+def init():
+    # Start ray instance
+    ray.init(num_cpus=1)
+
+    # Run test using this fixture
+    yield None
+
+    # Shutdown ray instance
+    ray.shutdown()
 
-@ray.remote
-def factorial(n):
-    if n == 0:
-        return 1
-    return n * ray.get(factorial.remote(n - 1))
 
+def test_recursion(init):
```

Comment:
@williamma12 does this test actually work in pytest?

Reply:
oops good catch. Did not intend to move the remote function inside the test and undoing it made the test pass.

Reply:
For consistency, can you rename `init` to `ray_start`?